ReText/editor.py

```diff
@@ -408,15 +438,72 @@ def leaveEvent(self, event):
 		palette = self.palette()
 		palette.setColor(QPalette.Window, colorValues['infoArea'])
 		palette.setColor(QPalette.WindowText,
-			self.editor.palette().color(QPalette.WindowText))
+						 self.editor.palette().color(QPalette.WindowText))
+		self.setPalette(palette)
+
+	def mousePressEvent(self, event):
+		pos = self.mapToParent(event.pos())
+		pos.setX(pos.x() - self.editor.lineNumberAreaWidth())
+		newEvent = QMouseEvent(event.type(), pos,
+							   event.button(), event.buttons(),
+							   event.modifiers())
+		self.editor.mousePressEvent(newEvent)
+
+
+class StatsArea(QLabel):
+	def __init__(self, editor):
+		QWidget.__init__(self, editor)
+		self.editor = editor
+		self.editor.cursorPositionChanged.connect(self.updateTextAndGeometry)
+		self.updateTextAndGeometry()
+		self.setAutoFillBackground(True)
+		palette = self.palette()
+		palette.setColor(QPalette.Window, colorValues['statsArea'])
+		self.setPalette(palette)
+		self.setCursor(Qt.IBeamCursor)
+
+	def updateTextAndGeometry(self):
+		text = self.getText()
+		self.setText(text)
+		viewport = self.editor.viewport()
+		metrics = self.fontMetrics()
+		width = metrics.width(text)
+		height = metrics.height()
+		self.resize(width, height)
+		leftSide = self.editor.lineNumberAreaWidth()
+		self.move(leftSide, viewport.height() - height)
+		self.setVisible(not globalSettings.useFakeVim)
+
+	def getText(self):
+		if not globalSettings.documentStatsEnabled:
+			return
+		template = '%d w : %d a : %d c'
+		words, alphaNums, characters = self.editor.statistics
+		return template % (words, alphaNums, characters)
+
+	def enterEvent(self, event):
+		palette = self.palette()
+		windowColor = QColor(colorValues['statsArea'])
+		windowColor.setAlpha(0x20)
+		palette.setColor(QPalette.Window, windowColor)
+		textColor = palette.color(QPalette.WindowText)
+		textColor.setAlpha(0x20)
+		palette.setColor(QPalette.WindowText, textColor)
+		self.setPalette(palette)
+
+	def leaveEvent(self, event):
+		palette = self.palette()
+		palette.setColor(QPalette.Window, colorValues['statsArea'])
+		palette.setColor(QPalette.WindowText,
+						 self.editor.palette().color(QPalette.WindowText))
 		self.setPalette(palette)
 
 	def mousePressEvent(self, event):
 		pos = self.mapToParent(event.pos())
 		pos.setX(pos.x() - self.editor.lineNumberAreaWidth())
 		newEvent = QMouseEvent(event.type(), pos,
-		                       event.button(), event.buttons(),
```

Comment:
This file uses tabs for indentation, but spaces for alignment. Please do not change this.